### PR TITLE
docs: Enable markdown linting and spell checking with related fixes

### PR DESCRIPTION
**Pull request by agneszitte**
GitHub Issue (If applicable): https://github.com/unoplatform/uno-private/issues/286

## PR Type

What kind of change does this PR introduce?

- Build or CI related changes
- Documentation content changes

## Description
- Enable markdown linting
- Enable spell-checking
- Adjust CI to have proper stages
- Fix related errors from the validations
- Fix a couple of additional list indentation issues
(Following changes for Uno core repo: https://github.com/unoplatform/uno/pull/16972)

*Uno Extensions documentation was built and tested locally to ensure there are no regressions with these related changes and that the proper fixes are correctly in place.

### CODE REVIEW

**Review by carldebilly — APPROVED** [review]
LGTM

**Comment by agneszitte** [comment]
> LGTM

Not done yet, still a draft. I am going through the last validations errors

File: doc/Learn/Http/HttpOverview.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ For more information about configuring `HttpClient` with custom endpoint options
 Similarly, **Refit endpoints** can be registered as services and configured in a similar way.
 
 ```csharp
-private IHost Host { get; }
+private IHost Host { get; set; }
```
Comment (jeromelaban) [inline doc/Learn/Http/HttpOverview.md:81]: Why is this needed?

**Review by jeromelaban — APPROVED** [review]

File: doc/Learn/Http/HttpOverview.md (hunk 1 above)
Comment (agneszitte) [reply doc/Learn/Http/HttpOverview.md:81]: @nickrandolph on some documentation pages we are setting a value for `Host`
image
Should make sense to have a setter for `Host` or just on the pages where we are setting a value for `Host`

Comment (carldebilly) [reply doc/Learn/Http/HttpOverview.md:81]:
```suggestion
private IHost Host { get; private set; }
```

File: build/ci/cspell.json [hunk 2]
```diff
@@ -0,0 +1,223 @@
+﻿{
+	"version": "0.2",
+	"language": "en",
+	"words": [
+		"Avalonia",
```
Comment (carldebilly) [inline build/ci/cspell.json:5]: We are specifying `Avalonia` somewhere?

**Review by carldebilly — APPROVED** [review]

File: build/ci/cspell.json (hunk 2 above)
Comment (agneszitte) [reply build/ci/cspell.json:5]: @carldebilly I based this file on the one we started on Uno for the first repo with spell-checking.
@jeromelaban it it worth adjusting the list based on the current repo only?

Comment (jeromelaban) [reply build/ci/cspell.json:5]: It's a common list, so that's fine.

File: doc/Learn/Http/HttpOverview.md (hunk 1 above)
Comment (jeromelaban) [reply doc/Learn/Http/HttpOverview.md:81]: Fine by me then

**Comment by agneszitte** [comment]
@Mergifyio backport release/stable/4.1

File: doc/Learn/Http/HttpOverview.md (hunk 1 above)
Comment (carldebilly) [reply doc/Learn/Http/HttpOverview.md:81]: Would prefer the `private set`.
